src/pages/register.tsx

```diff
@@ -2,11 +2,45 @@ import { RegisterRouter } from '~/navigation/router'
 import Layout from '~/components/layout'
 import SEO from '~/components/seo'
 import type { ReadonlyRouteComponentProps } from '~/util/readonly-types'
+import { useObservable, useObservableState } from 'observable-hooks'
+import { registrationCountdownCheck } from '~/apis/attsrv'
+import { EMPTY } from 'rxjs'
+import { catchError, map } from 'rxjs/operators'
+import { AjaxError } from 'rxjs/ajax'
+import { StatusCodes } from 'http-status-codes'
+import { InitiateLogin } from '~/state/actions/auth'
+import { useAppDispatch } from '~/hooks/redux'
+import NotOpenYet from '~/components/funnels/funnels/register/not-open-yet'
 
 export const Head = () => <SEO title="Register" />
 
-const RegisterPage = (_: ReadonlyRouteComponentProps) => <Layout>
-	{RegisterRouter()}
-</Layout>
+const Content = ({ isRegistrationOpen }: { readonly isRegistrationOpen: boolean | null }) => {
+	switch (isRegistrationOpen) {
+		case true: return <RegisterRouter/>
+		case false: return <NotOpenYet/>
+		case null: return <div>Loading...</div>
+	}
+}
+
+const RegisterPage = (_: ReadonlyRouteComponentProps) => {
+	const dispatch = useAppDispatch()
+	const countdownQuery$ = useObservable(() => registrationCountdownCheck().pipe(
+		map(result => result.response.countdown <= 0),
+		catchError(err => {
+			if (err instanceof AjaxError) {
+				if (err.status === StatusCodes.UNAUTHORIZED) {
```

Comment:
Note: This can happen during **any** of the ajax calls to the services.

If the user stays on the page for a while and either the token in the cookie or the cookie itself expires, which happens after about an hour, or 6 hours respectively, any of the ajax calls can return this error, and you should dispatch to login and set a return URL that brings the user back to their current place.

Reply:
I know, the more generic calls are handled elsewhere.